### ISSUE TITLE
[naming-convention] Allow "selector" to specify an array of selectors

### ISSUE THREAD

**Issue by octogonz** [typescript-eslint/typescript-eslint#2266]
**Repro**

Suppose I want to define an option block that applies to several selectors, which do not match up with one of the predefined "group selectors".  Currently the option block needs to be duplicated, like this:

```JS
'@typescript-eslint/naming-convention': [
    'error',
  {
    selector: 'class',
    format: ['PascalCase'],
    leadingUnderscore: 'allow'
  },
  {
    selector: 'typeAlias',
    format: ['PascalCase'],
    leadingUnderscore: 'allow'
  },
  {
    selector: 'enum',
    format: ['PascalCase'],
    leadingUnderscore: 'allow'
  },
  {
    selector: 'typeParameter',
    format: ['PascalCase'],
    leadingUnderscore: 'allow'
  },
  . . .
]
```

**Expected Result**

It should be possible to avoid duplication by specifying an array of selectors, like this:

```JS
'@typescript-eslint/naming-convention': [
    'error',
   {
    // Group selector for: class, interface, typeAlias, enum, typeParameter
    selectors: ['class', 'typeAlias', 'enum', 'typeParameter'],
    format: ['PascalCase'],
    leadingUnderscore: 'allow'
  },
  . . .
]
```

**Additional Info**

This arose in a real world situation with [this option block](https://github.com/microsoft/rushstack/blob/e1428a0ad6b576f55b0f6982f4bfed40cacac180/stack/eslint-config/index.js#L292) and a number of other cases in that same file.

I solved it by creating a macro [expandNamingConventionSelectors()](https://github.com/microsoft/rushstack/blob/e1428a0ad6b576f55b0f6982f4bfed40cacac180/stack/eslint-config/macros.js#L63) that expands the `selectors` array into multiple blocks with different `selector` fields.

This worked, however it is clumsy. Also, one of our repos creates a test snapshot with the expanded eslintrc object, and the macro output is difficult to read after it is expanded. 

It would be better if `naming-convention` had built-in support for array of selectors, without need for a macro to generate the blocks.  I can't think of any design problems with allowing that.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.3.0` |
| `@typescript-eslint/parser`        | `3.3.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `7.2.0` |
| `node`                             | `12.17.0` |
| `npm`                              | `6.14.4` |

**Comment by octogonz** [comment]
BTW an explicit array of selectors might also be more readable for people who work on lint rules. An explicit list avoids the need to consult the documentation and remember what each [group selector name](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md#group-selectors) is referring to.

**Comment by octogonz** [comment]
Awesome, thank you @GavinWu1991 and @bradzacher !
